src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/profiler/ProfilerEditingTarget.java

```diff
@@ -393,10 +486,125 @@ public void onError(ServerError error)
+               continuation.execute(null);
             }
          });
    }
+   
+   private void persistHtmlPath(String htmlPath)
+   {
+      HashMap<String, String> props = new HashMap<String, String>();
+      props.put("htmlPath", htmlPath);
+      
+      sourceServer_.modifyDocumentProperties(
+         doc_.getId(),
+         props,
+         new SimpleRequestCallback<Void>("Error")
+         {
+            @Override
+            public void onResponseReceived(Void response)
+            {
+            }
+
+            @Override
+            public void onError(ServerError error)
+            {
+               globalDisplay_.showErrorMessage("Failed to Save Profile Properties",
+                     error.getMessage());
+            }
+      });
+   }
+
+   private void saveNewFile(final String suggestedPath,
+                            final Command executeOnSuccess)
+   {
+      FileSystemItem fsi;
+      if (suggestedPath != null)
+         fsi = FileSystemItem.createFile(suggestedPath);
+      else
+         fsi = workbenchContext_.getDefaultFileDialogDir();
+ 
+      fileDialogs_.saveFile(
+            "Save File - " + getName().getValue(),
+            fileContext_,
+            fsi,
+            fileType_.getDefaultExtension(),
+            false,
+            new ProgressOperationWithInput<FileSystemItem>()
+            {
+               public void execute(final FileSystemItem saveItem,
+                                   final ProgressIndicator indicator)
+               {
+                  if (saveItem == null)
+                     return;
+
+                  workbenchContext_.setDefaultFileDialogDir(
+                        saveItem.getParentPath());
+                  
+                  final String toPath = saveItem.getPath();
+                  server_.copyProfile(
+                     getPath(),
+                     toPath,
+                     new ServerRequestCallback<JavaScriptObject>() {
+                        @Override
+                        public void onResponseReceived(JavaScriptObject response)
+                        {
+                           eventBus_.fireEvent(new FileEditEvent(saveItem));
+                           
+                           indicator.onCompleted();
+                        }
+
+                        @Override
+                        public void onError(ServerError error)
+                        {
+                           indicator.onCompleted();
+                           globalDisplay_.showErrorMessage("Failed to Save Profile",
+                                 error.getMessage());
+                        }
+                  });
+               }
+            });
+   }
 
+   private Command postSaveProfileCommand()
+   {
+      return new Command()
+      {
+         public void execute()
+         {
+         }
+      };
+   }
+   
+   private static void onMessage(String message, String file, int line)
+   {  
+      if (message == "sourcefile")
```

Comment:
In Java, `==` on Strings checks equality of references rather than equality of values -- this gets lost after a GWT transpile, though.

I may just be bike-shedding though :)